build.sbt

```diff
@@ -4,49 +4,75 @@ lazy val `scala-2.12`     = "2.12.12"
+    headerLicense  := Some(HeaderLicense.Custom(
+      """|Copyright (c) 2020-2021 by Rob Norris
+        |This software is licensed under the MIT License (MIT).
+        |For more information see LICENSE or https://opensource.org/licenses/MIT
+        |""".stripMargin
+      )
+    ),
+
+    // Compilation
+    scalaVersion       := `scala-2.13`,
+    crossScalaVersions := Seq(`scala-2.12`, `scala-2.13`, `scala-3.0`),
+    Compile / doc     / scalacOptions --= Seq("-Xfatal-warnings"),
+    Compile / doc     / scalacOptions ++= Seq(
+      "-groups",
+      "-sourcepath", (LocalRootProject / baseDirectory).value.getAbsolutePath,
+      "-doc-source-url", "https://github.com/tpolecat/sourcepos/blob/v" + version.value + "€{FILE_PATH}.scala",
+    ),
+
+    // MUnit
+    Compile / unmanagedSourceDirectories ++= {
+      val major = if (scalaVersion.value.startsWith("3")) "-3" else "-2"
+      List(CrossType.Pure, CrossType.Full).flatMap(
+        _.sharedSrcDir(baseDirectory.value, "main").toList.map(f => file(f.getPath + major))
+      )
+    },
+    Test / unmanagedSourceDirectories ++= {
+      val major = if (scalaVersion.value.startsWith("3")) "-3" else "-2"
+      List(CrossType.Pure, CrossType.Full).flatMap(
+        _.sharedSrcDir(baseDirectory.value, "test").toList.map(f => file(f.getPath + major))
+      )
+    },
+    libraryDependencies += ("org.scalameta" %%% "munit" % "0.7.27" % Test),
+
+    testFrameworks += new TestFramework("munit.Framework"),
+
+    // dottydoc really doesn't work at all right now
+    Compile / doc / sources := {
+      val old = (Compile / doc / sources).value
+      if (scalaVersion.value.startsWith("3."))
+        Seq()
+      else
+        old
+    },
+  )
+  .settings(
+    libraryDependencies ++= {
+      if (scalaVersion.value.startsWith("3")) Nil else List("org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided")
+    }
   )
-)
-
-// Compilation
-scalaVersion       := `scala-2.13`
-crossScalaVersions := Seq(`scala-2.12`, `scala-2.13`, `scala-3.0`)
-Compile / doc     / scalacOptions --= Seq("-Xfatal-warnings")
-Compile / doc     / scalacOptions ++= Seq(
-  "-groups",
-  "-sourcepath", (LocalRootProject / baseDirectory).value.getAbsolutePath,
-  "-doc-source-url", "https://github.com/tpolecat/sourcepos/blob/v" + version.value + "€{FILE_PATH}.scala",
-)
-
-// MUnit
-libraryDependencies += "org.scalameta" %% "munit" % "0.7.26" % Test
-testFrameworks += new TestFramework("munit.Framework")
-
-// Scala 2 needs scala-reflect
-libraryDependencies ++= Seq("org.scala-lang" % "scala-reflect" % scalaVersion.value).filterNot(_ => scalaVersion.value.startsWith("3."))
-
-// dottydoc really doesn't work at all right now
-Compile / doc / sources := {
-  val old = (Compile / doc / sources).value
-  if (scalaVersion.value.startsWith("3."))
-    Seq()
-  else
-    old
-}
-
-enablePlugins(AutomateHeaderPlugin)
+  .jsSettings(
+    scalaJSLinkerConfig ~= (_.withModuleKind(ModuleKind.CommonJSModule))
```

Comment:
Not 100% on this setting

Reply:
This is fine! It enables `munit` to import filesystem modules in Node.js. In fact you could make it `Test / scalaJSLinkerConfig`.

Reply:
Actually if you update to the latest munit this shouldn't be needed anymore.